### PR TITLE
chore: replace gorilla/mux with go-chi/chi

### LINKED ISSUE #301: migrate from `gorilla/mux` to `go-chi/chi`
See #300 for description.

## Acceptance criteria

- [x] all currently registered `gorilla/mux` routes replaced by `go-chi/chi`
- [x] regression all tests still passing without changes in execution
- [x] no `github.com/gorilla/mux` module in the `go.mod` remaining

### PR DESCRIPTION
**Pull request by muhlemmer**
This PR refactors the use `go-chi/chi` as router, instead of `gorilla/mux`.

Closes #301

### CODE REVIEW

File: pkg/op/auth_request_test.go [hunk 1]
```diff
@@ -12,7 +12,6 @@ import (
 	"github.com/gorilla/schema"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-
 	httphelper "github.com/zitadel/oidc/v2/pkg/http"
 	"github.com/zitadel/oidc/v2/pkg/oidc"
 	"github.com/zitadel/oidc/v2/pkg/op"
```
Comment (adlerhurst) [inline pkg/op/auth_request_test.go:17]: is the branch for service accounts mentioned on line 21 merged? if so can we remove the comments and execute the tests?

File: go.mod [hunk 2]
```diff
@@ -3,10 +3,10 @@ module github.com/zitadel/oidc/v2
 go 1.18
 
 require (
+	github.com/go-chi/chi v1.5.4
 	github.com/golang/mock v1.6.0
 	github.com/google/go-github/v31 v31.0.0
 	github.com/google/uuid v1.3.0
-	github.com/gorilla/mux v1.8.0
 	github.com/gorilla/schema v1.2.0
 	github.com/gorilla/securecookie v1.1.1
```
[the remark this answers is not in the document]
Comment (muhlemmer) [reply go.mod:11]: Yes, It does make sense. But I haven't found a good replacement yet. But essentially it is about encoding and signing a payload, much like we do with tokens.

File: pkg/op/auth_request_test.go (hunk 1 above)
Comment (muhlemmer) [reply pkg/op/auth_request_test.go:17]: I will investigate and create a new PR for that. The git blame is 2 years old from Livio ;)

File: go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (muhlemmer) [reply go.mod:11]: Note, it is already mentioned in the epic https://github.com/zitadel/oidc/issues/300#issue-1605552940.

[the remark this answers is not in the document]
Comment (adlerhurst) [reply go.mod:11]: perfect

File: pkg/op/auth_request_test.go (hunk 1 above)
Comment (adlerhurst) [reply pkg/op/auth_request_test.go:17]: perfect, i approve this one

**Review by adlerhurst — APPROVED** [review]

File: pkg/op/auth_request_test.go (hunk 1 above)
Comment (muhlemmer) [reply pkg/op/auth_request_test.go:17]: https://github.com/zitadel/oidc/issues/335

**Pull request merged**